test/integration/test_data_manager_table_reload.py

```diff
@@ -0,0 +1,134 @@
+FETCH_GENOME_DBKEYS_ALL_FASTA_INPUT = {"dbkey_source|dbkey_source_selector": "new",
+                                       "dbkey_source|dbkey": "NC_001617.1",
+                                       "dbkey_source|dbkey_name": "NC_001617.1",
+                                       "sequence_name": "NC_001617.1",
+                                       "sequence_id": "NC_001617.1",
+                                       "reference_source|reference_source_selector": "ncbi",
+                                       "reference_source|requested_identifier": "NC_001617.1",
+                                       "sorting|sort_selector": "as_is"}
+SAM_FASTA_ID = "toolshed.g2.bx.psu.edu/repos/devteam/data_manager_sam_fasta_index_builder/sam_fasta_index_builder/0.0.2"
+SAM_FASTA_INPUT = {"all_fasta_source": "NC_001617.1", "sequence_name": "", "sequence_id": ""}
+
+
+class DataManagerIntegrationTestCase(integration_util.IntegrationTestCase):
+
+    """Test data manager installation and table reload through the API"""
+
+    framework_tool_and_types = True
+
+    def setUp(self):
+        super(DataManagerIntegrationTestCase, self).setUp()
+        self.dataset_populator = DatasetPopulator(self.galaxy_interactor)
+
+    @classmethod
+    def handle_galaxy_config_kwds(cls, config):
+        try:
+            import watchdog  # noqa: F401
+        except ImportError:
+            raise SkipTest("watchdog library is not available")
+        cls.username = cls.get_secure_ascii_digits()
+        cls.conda_tmp_prefix = tempfile.mkdtemp()
+        cls.shed_tools_dir = tempfile.mkdtemp()
+        cls.shed_tool_data_dir = tempfile.mkdtemp()
+        config["conda_auto_init"] = True
+        config["conda_auto_install"] = True
+        config["conda_prefix"] = os.path.join(cls.conda_tmp_prefix, 'conda')
+        config["tool_sheds_config_file"] = TOOL_SHEDS_CONF
+        config["tool_config_file"] = os.path.join(cls.shed_tools_dir, 'shed_tool_conf.xml')
+        config["shed_data_manager_config_file"] = os.path.join(cls.shed_tool_data_dir, 'shed_data_manager_config_file')
+        config["shed_tool_data_table_config"] = os.path.join(cls.shed_tool_data_dir, 'shed_data_table_conf.xml')
+        config["shed_tool_data_path"] = cls.shed_tool_data_dir
+        config["tool_data_path"] = cls.shed_tool_data_dir
+        config["watch_tool_data_dir"] = True
+        config["admin_users"] = "%s@galaxy.org" % cls.username
+        with open(config["tool_config_file"], 'w') as tool_conf_file:
+            tool_conf_file.write(SHED_TOOL_CONF.substitute(shed_tools_path=cls.shed_tools_dir))
+        with open(config["shed_data_manager_config_file"], 'w') as shed_data_config:
+            shed_data_config.write(SHED_DATA_MANAGER_CONF)
+        with open(config["shed_tool_data_table_config"], 'w') as shed_data_table_config:
+            shed_data_table_config.write(SHED_DATA_TABLES)
+
+    @classmethod
+    def tearDownClass(cls):
```

Comment:
How about instead of overriding this you put and empty method called ``_cleanup_galaxy()`` that matches ``_prepare_galaxy()`` in the base class and then just use that here. Might be even cleaner to define a list in parent class called ``_configuration_files`` that you can just append to that will get cleaned up automatically. This way you don't need to do anything in this class besides register them and then later on we can make the tests respect ``GALAXY_TEST_NO_CLEANUP``.

Reply:
Actually I like this idea so much I already implemented it. If you just append these directories to ``self._test_driver.temp_directories`` they should be taken care of automatically. No need to declare a tearDown method.